src/components/ClaimENSPage/ClaimENSPage.tsx

```diff
@@ -76,14 +77,26 @@ export default class ClaimENSPage extends React.PureComponent<Props, State> {
     this.setState({ isLoading: false })
   }
 
-  handleClaim = () => {
+  handleClaim = async () => {
     const { onOpenModal } = this.props
     const { name } = this.state
-    onOpenModal('ClaimNameFatFingerModal', { originalName: name })
+    this.setState({ isLoading: true })
+    const isAvailable = await isNameAvailable(name)
```

Comment:
What do you think of adding a `try/catch` clause here? If for some reason `isNameAvailable` throws, the UI will keep loading forever and this _check_ is not a 100% blocker for the flow since the blockchain has the final word

Reply:
good eye! I'll make it.